doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -0,0 +1,500 @@
+
+##### Config interface type
+
+Configuring interface type takes effect only if auto negotiation is disabled. If auto negotiation is enabled, this command still saves interface type value to CONFIG_DB.
+
+```
+Format:
+  config interface type <interface_name> <interface_type>
+
+Arguments:
+  interface_name: name of the interface to be configured. e.g: Ethernet0
+  interface_type: interface type, valid value include: KR4, SR4 and so on. A list of valid interface type could be found at saiport.h.
+
+Example:
+  config interface type Ethernet0 KR4
+
+Return:
+  error message if interface_name or interface_type is invalid otherwise empty
+```
+
+##### Config advertised interface types
+
+Configuring advertised interface types takes effect only if auto negotiation is enabled. If auto negotiation is disabled, this command still saves advertised interface types value to CONFIG_DB.
+
+```
+Format:
+  config interface advertised-types <interface_name> <interface_type_list>
+
+Arguments:
+  interface_name: name of the interface to be configured. e.g: Ethernet0
+  interface_type_list: a list of interface types to be advertised or "all". e.g: KR4,SR4.
+
+Example:
+  config interface advertised-types Ethernet0 KR4,SR4
+  config interface advertised-types all
+
+Return:
+  error message if interface_name or interface_type_list is invalid otherwise empty
+
+Note:
+  interface_type_list value "all" means all supported interface type 
+```
+
+This command always replace the advertised interface types instead of append. For example, say the current advertised interface types value are "KR4,SR4", if user configure it with `config interface advertised-types Ethernet0 CR4`, the advertised interface types value will be changed to "CR4".
+
+##### Show interfaces auto negotiation status
+
+As command `show interfaces status` already has 11 columns, a new CLI command will be added to display the port auto negotiation status. All data of this command are fetched from **APPL_DB**.
+
+```
+Format:
+  show interfaces auto-neg-status <interface_name>
+
+Arguments:
+  interface_name: optional. Name of the interface to be shown. e.g: Ethernet0. If interface_name is not given, this command shows auto negotiation status for all interfaces.
+
+Example:
+  show interfaces auto-neg-status
+  show interfaces auto-neg-status Ethernet0
+
+Return:
+  error message if interface_name is invalid otherwise:
+
+  Interface    Auto-Neg Mode    Speed    Adv Speeds    Type    Adv Types
+-----------  ---------------  -------  ------------  ------  -----------
+  Ethernet0          enabled     100G      40G,100G     N/A      CR4,KR4
+ Ethernet32         disabled      40G           N/A     N/A          N/A
+```
+
+#### Config DB Enhancements  
+
+SONiC already defined two fields related to port speed setting: **speed**, **autoneg**. 3 new fields **adv_speeds**, **interface_type**, **adv_interface_types** will be added to **PORT** table:
```

Comment:
since there is enhancement on the config db, we need to add this new fields in the sonic yang model.

Reply:
@praveen-li , comments?

Reply:
@lguohan : if we decided to add these fields in the PORT table of ConfigDB, if es, then these must be added to the yang models. Else breakout will not be supported and other config validation will also be unsuccessful.

Reply:
as discussed in the sonic yang subgroup, all new hld will need sub pr to add yang model in src/sonic-yang-model, the hld approved is contingent on the approval of yang model pr.

Reply:
Hi @lguohan and @praveen-li , I created a PR for the yang modle change, please help review it. Thanks.
https://github.com/Azure/sonic-buildimage/pull/6948